cipher/rsa/rsa2.go

```diff
@@ -0,0 +1,131 @@
+/*
+rsa2.go
+description: RSA encryption and decryption including key generation
+details: [RSA wiki](https://en.wikipedia.org/wiki/RSA_(cryptosystem))
+author(s): [ddaniel27](https://github.com/ddaniel27)
+*/
+package rsa
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"math/rand"
+
+	"github.com/TheAlgorithms/Go/math/gcd"
+	"github.com/TheAlgorithms/Go/math/lcm"
+	"github.com/TheAlgorithms/Go/math/modular"
+	"github.com/TheAlgorithms/Go/math/prime"
+)
+
+// rsa struct contains the public key, private key and modulus
+type rsa struct {
+	publicKey  uint64
+	privateKey uint64
+	modulus    uint64
+}
+
+// InitRSA initializes the RSA algorithm
+// returns the RSA object
+func InitRSA() *rsa {
```

Comment:
These kinds of functions usually have the naming convention `New<Thing>`, due to the similar behaviour of the `new` builtin function. However when `<Thing>` is the package name, just `New` also works.

Reply:
Makes sense.